test/vfd_swmr_indep_rw_writer.c

```diff
@@ -0,0 +1,788 @@
+                    HDfprintf(stderr, "-%c argument %lu too large", ch, tmp);
+                    TEST_ERROR;
+                }
+                if ((ch == 'c' || ch == 'r') && tmp == 0) {
+                    HDfprintf(stderr, "-%c argument %lu must be >= 1", ch, tmp);
+                    TEST_ERROR;
+                }
+
+                if (ch == 'c')
+                    s->cols = (unsigned)tmp;
+                else if (ch == 'r')
+                    s->rows = (unsigned)tmp;
+                else if (ch == 't')
+                    s->tick_len = (unsigned)tmp;
+                else if (ch == 'm')
+                    s->max_lag = (unsigned)tmp;
+                else if (ch == 'B')
+                    s->pbs = (unsigned)tmp;
+                else if (ch == 's')
+                    s->ps = (unsigned)tmp;
+                else if (ch == 'u')
+                    s->check_interval = (unsigned)tmp;
+                break;
+            case 'q':
+                verbosity = 0;
+                break;
+            case '?':
+            default:
+                usage(s->progname);
+                break;
+        }
+    }
+    argc -= optind;
+    argv += optind;
+
+    if (argc > 0) {
+        HDprintf("unexpected command-line arguments\n");
+        TEST_ERROR;
+    }
+
+    s->filename[0] = "vfd_swmr_indep_wr_p0.h5";
+    s->filename[1] = "vfd_swmr_indep_wr_p1.h5";
+    s->dims[0]     = s->rows;
+    s->dims[1]     = s->cols;
+
+    personality = HDstrstr(s->progname, "vfd_swmr_indep_wr");
+
+    if (personality != NULL && HDstrcmp(personality, "vfd_swmr_indep_wr_p0") == 0)
+        s->first_proc = true;
+    else if (personality != NULL && HDstrcmp(personality, "vfd_swmr_indep_wr_p1") == 0)
+        s->first_proc = false;
+    else {
+        HDprintf("unknown personality, expected vfd_swmr_indep_wr_{p0,p1}\n");
+        TEST_ERROR;
+    }
+
+    return true;
+
+error:
+    if (tfile)
+        HDfree(tfile);
+    return false;
+}
+
+static bool
+indep_init_vfd_swmr_config_plist(state_t *s, bool writer, const char *mdf_path)
+{
+
+    H5F_vfd_swmr_config_t config;
+
+    /* config, tick_len, max_lag, writer, flush_raw_data, md_pages_reserved, md_file_path */
+    init_vfd_swmr_config(&config, s->tick_len, s->max_lag, writer, TRUE, 128, mdf_path);
+
+    /* Pass the use_vfd_swmr, only_meta_page, page buffer size, config to vfd_swmr_create_fapl().*/
+    if ((s->fapl = vfd_swmr_create_fapl(true, s->use_vfd_swmr, true, s->pbs, &config)) < 0) {
+        printf("vfd_swmr_create_fapl failed\n");
+        TEST_ERROR;
+    }
+
+    /* Set fs_strategy (file space strategy) and fs_page_size (file space page size) */
+    if ((s->fcpl = vfd_swmr_create_fcpl(H5F_FSPACE_STRATEGY_PAGE, s->ps)) < 0) {
+        HDprintf("vfd_swmr_create_fcpl() failed");
+        TEST_ERROR;
+    }
+
+    return true;
+
+error:
+    return false;
+}
+
+static bool
+write_dataset(const state_t *s, mat_t *mat)
+{
+
+    char         dname[sizeof("/dataset-?")];
```

Comment:
Will do

Reply:
Since the dataset name will be either /dataset-0 or /dataset-1 for the H5Dcreate2(), I prefer to use the current way instead of a fixed MACRO, which I need to write H5Dcreate2() twice.